rust/parquet/src/arrow/schema.rs

```diff
@@ -1430,12 +1491,66 @@ mod tests {
                 ),
                 Field::new("c32", DataType::LargeBinary, true),
                 Field::new("c33", DataType::LargeUtf8, true),
+                // Field::new(
+                //     "c34",
+                //     DataType::LargeList(Box::new(DataType::List(Box::new(
+                //         DataType::Struct(vec![
+                //             Field::new("a", DataType::Int16, true),
+                //             Field::new("b", DataType::Float64, true),
+                //         ]),
+                //     )))),
+                //     true,
+                // ),
+            ],
+            metadata,
+        );
+
+        // write to an empty parquet file so that schema is serialized
+        let file = get_temp_file("test_arrow_schema_roundtrip.parquet", &[]);
+        let mut writer = ArrowWriter::try_new(
+            file.try_clone().unwrap(),
+            Arc::new(schema.clone()),
+            None,
+        )?;
+        writer.close()?;
+
+        // read file back
+        let parquet_reader = SerializedFileReader::try_from(file)?;
+        let mut arrow_reader = ParquetFileArrowReader::new(Rc::new(parquet_reader));
+        let read_schema = arrow_reader.get_schema()?;
+        assert_eq!(schema, read_schema);
+
+        // read all fields by columns
+        let partial_read_schema =
+            arrow_reader.get_schema_by_columns(0..(schema.fields().len()), false)?;
```

Comment:
@carols10cents I changed this to read all columns by root.

Reply:
Nice! I hadn't yet figured out why there were a different number of Parquet columns and Arrow fields; I think I'm starting to understand now :)